### ISSUE TITLE
Plugin E2E: Add support for accessibility testing

### ISSUE THREAD

**Issue by sunker** [grafana/plugin-tools#2457]
### Which areas does this feature request relate to

- [ ] Create Plugin
- [ ] Sign Plugin
- [x] Plugin E2E
- [ ] Plugin Meta Extractor
- [ ] Documentation

### Problem

Plugin developers have no built-in way to test accessibility in their plugins using plugin-e2e. A11y violations can slip into production without automated checks and developers shouldn't need to wire up their own AxeBuilder integration.

### Solution

Suggested solution is to add:

1. A fixture that wraps Playwright's [AxeBuilder](https://playwright.dev/docs/accessibility-testing#scanning-an-entire-page) to scan for a11y violations. It should support scanning the full page or a specific selector, so plugin devs can restrict scans to the areas their plugin controls (e.g. query editor only, not the full Grafana UI).

2. A [custom matcher](https://playwright.dev/docs/test-assertions#add-custom-matchers-using-expectextend) for readable assertions with clear, actionable error output.

```ts
import { expect, test } from '@grafana/plugin-e2e';

test('query editor should have no a11y violations', async ({ scanForA11yViolations, panelEditPage}) => {
  const results = await scanForA11yViolations({
    include: panelEditPage.getQueryEditorRow('A'),
  });
  expect(results).toHaveNoA11yViolations();
});
```

**Comment by fastfrwrd** [comment]
Anything we do will need to be able to hook up a report artifact, so as long as the custom matcher updates the test annotations with the raw axe report (as it does in the initial PR), we should be able to build that reporter in the grafana/grafana repo.

**Comment by sunker** [comment]
Fixed in https://github.com/grafana/plugin-tools/pull/2462
